core-tests/shared/src/test/scala/zio/FiberRefSpec.scala

```diff
@@ -440,6 +440,39 @@ object FiberRefSpec extends ZIOBaseSpec {
              }
         value <- promise.await
       } yield assertTrue(value)
+    },
```

Comment:
Let's have a test case for the nephew case.
Thanks for the quick work on this

Reply:
Done now!